iml-orm/src/lib.rs

```diff
@@ -75,10 +83,71 @@ pub use tokio_diesel;
 #[cfg(feature = "postgres-interop")]
 pub use r2d2;
 
+#[derive(Debug, Error)]
+pub enum ImlOrmError {
+    #[error(transparent)]
+    R2D2Error(#[from] r2d2::Error),
+    #[error(transparent)]
+    OneshotCanceled(#[from] oneshot::Canceled),
+    #[error(transparent)]
+    AsyncError(#[from] tokio_diesel::AsyncError),
+}
+
+#[cfg(feature = "warp-filters")]
+impl warp::reject::Reject for ImlOrmError {}
+
 #[cfg(feature = "postgres-interop")]
 /// Get a new connection pool based on the envs connection string.
-pub fn pool() -> Result<DbPool, r2d2::Error> {
+pub fn pool() -> Result<DbPool, ImlOrmError> {
     let manager = ConnectionManager::<PgConnection>::new(get_db_conn_string());
 
-    Pool::builder().build(manager)
+    Pool::builder()
+        .build(manager)
+        .map_err(ImlOrmError::R2D2Error)
+}
+
+#[cfg(feature = "warp-filters")]
+type PoolSender = oneshot::Sender<DbPool>;
+
+#[cfg(feature = "warp-filters")]
+pub fn get_cloned_pool(
+    pool: DbPool,
+) -> (mpsc::UnboundedSender<PoolSender>, impl Future<Output = ()>) {
+    let (tx, rx) = mpsc::unbounded();
+
+    let fut = rx.for_each(move |sender: PoolSender| {
+        let _ = sender
+            .send(pool.clone())
```

Comment:
Does the pool get cloned on each request?
Isn't this too expensive?

Reply:
The type for pool is: `pub struct Pool<M>(Arc<SharedPool<M>>)`

`.clone()` is incrementing the refcount of the `Arc`, so this is very cheap.